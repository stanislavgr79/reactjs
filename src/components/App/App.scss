.footer {
    p {
        margin: auto;
        text-align: center;
    }
}

.site_name {
    font-size: 22px;
}