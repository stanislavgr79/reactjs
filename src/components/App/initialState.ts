const initialState = {
  message: 'task2',
  time: new Date().toLocaleTimeString('en-US'),
};

export default initialState;
