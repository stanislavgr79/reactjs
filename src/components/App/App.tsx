import React from 'react';
import { HashRouter, Switch, Route, Redirect } from 'react-router-dom';
import { Provider } from 'react-redux';
import { rootReducer } from '../../redux/redux-store';

import Intro from '../../pages/Intro';
import './App.scss';

const App = () => {
  return (
    <HashRouter>
      <Provider store={rootReducer}>
        <Switch>
          <Route exact key="Intro" path="/" render={() => <Intro />} />
          <Redirect key="RedirectIntro" to="/" />
        </Switch>
      </Provider>
    </HashRouter>
  );
};

export default React.memo(App);
