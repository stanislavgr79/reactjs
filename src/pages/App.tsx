import React from 'react';
import { Switch, Route, Router, BrowserRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import Intro from '../pages/Intro';
import './App.scss';
import NoMatch from '../pages/NoMatch';
import { Provider } from 'react-redux';
import store from '../redux/redux-store';

const history = createMemoryHistory();

const App = () => {
  return (
    <Router history={history}>
      <Provider store={store}>
        <Switch>
          <Route exact key="Home" path="/" render={() => <Intro />} />
          <Route exact key="Intro" path="/movies" render={() => <Intro />} />
          <Route exact key="Movie" path="/movies/:id" render={() => <Intro />} />
          <Route exact key="404" path="/404" render={() => <NoMatch />} />
          <Route path="*">
            <NoMatch />
          </Route>
        </Switch>
      </Provider>
    </Router>
  );
};

export default React.memo(App);
