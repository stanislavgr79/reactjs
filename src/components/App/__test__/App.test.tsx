import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import App from '../App';

jest.mock('../../../pages/Intro', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="intro"></div>;
    },
  };
});
jest.mock('../../../pages/NoMatch', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="noMatch"></div>;
    },
  };
});

describe('App', () => {
  const rrd = require('react-router-dom');
  rrd.BrowserRouter = ({ children }: any) => <div>{children}</div>;

  test('render intro page with `/`', () => {
    const tree = render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>,
    );

    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render intro page with `/movies`', () => {
    const tree = render(
      <MemoryRouter initialEntries={['/movies']}>
        <App />
      </MemoryRouter>,
    );

    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render intro page with `/movies/12345`', () => {
    const tree = render(
      <MemoryRouter initialEntries={['/movies/12345']}>
        <App />
      </MemoryRouter>,
    );

    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render noMatch 404 page with `/404`', () => {
    const tree = render(
      <MemoryRouter initialEntries={['/404']}>
        <App />
      </MemoryRouter>,
    );

    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render noMatch 404 page with bad url', () => {
    const tree = render(
      <MemoryRouter initialEntries={['/random']}>
        <App />
      </MemoryRouter>,
    );

    expect(tree.baseElement).toMatchSnapshot();
  });
});
