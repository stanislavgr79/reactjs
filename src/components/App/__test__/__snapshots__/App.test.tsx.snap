// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render intro page with \`/\` 1`] = `
<body>
  <div>
    <div>
      <div
        class="intro"
      />
    </div>
  </div>
</body>
`;

exports[`App render intro page with \`/movies/12345\` 1`] = `
<body>
  <div>
    <div>
      <div
        class="intro"
      />
    </div>
  </div>
</body>
`;

exports[`App render intro page with \`/movies\` 1`] = `
<body>
  <div>
    <div>
      <div
        class="intro"
      />
    </div>
  </div>
</body>
`;

exports[`App render noMatch 404 page with \`/404\` 1`] = `
<body>
  <div>
    <div>
      <div
        class="noMatch"
      />
    </div>
  </div>
</body>
`;

exports[`App render noMatch 404 page with bad url 1`] = `
<body>
  <div>
    <div>
      <div
        class="noMatch"
      />
    </div>
  </div>
</body>
`;
